lib/feedsme/index.js

```diff
@@ -54,153 +64,223 @@ function pkgName(pkg) {
+};
 
-  function finish(err) {
-    if (err) return done(err);
+Feedsme.prototype.triggerDepOf = async function triggerDepOf({ pkg, depOf }) {
+  if (!depOf || (depOf && !depOf.dependentOf)) return this.log.info(`${pkg.name} does not have a parent package it depends on`);
 
-    app.log.info(`Successfully processed change for ${ pkgName(pkg) } in ${ env }`, meta);
-    done.apply(done, arguments);
-  }
+  // Fetch the release line for the parent package that owns it
+  const line = await this.release.get({ pkg: depOf.dependentOf });
 
-  app.log.info(`Processing change for ${ pkgName(pkg) } in ${ env }`, meta);
-  async.parallel([
-    this.trigger.bind(this, env, data),
-    this.resolve.bind(this, env, pkg)
-  ], finish);
+  await this.release.dependent.add({
+    pkg: line.pkg,
+    version: line.version,
+    dependent: pkg.name,
+    dependentVersion: pkg.version
+  });
 };
 
+
 /**
- * Trigger new builds for all the stored dependencies.
+ * Trigger new builds for all the stored dependents.
  *
- * @param {String} env Environment for the builds.
- * @param {Object} data Package.json of the latest build.
- * @param {Function} done Completion callback
+ * @param {Object} options Options for triggering all dependents
+ * @returns {Promise} to resolve
  * @api private
  */
-Feedsme.prototype.trigger = function trigger(env, data, done) {
-  var feedsme = this;
-  var app = feedsme.app;
-  var BuildHead = app.models.BuildHead;
-  var Dependent = app.models.Dependent;
-  var Package = app.models.Package;
-  var Version = app.models.Version;
-  var npm = app.config.get('npm');
-  var name = data.name;
-
-  app.log.debug(`Trying to rebuild the dependents of ${ name }`);
-
-  Dependent.get(name, function found(error, details) {
-    if (error || !details) {
-      if (error) {
-        app.log.debug(`Failed to retrieve dependents of ${ name }: ${ error.message }`, {
-          innerErrors: error.innerErrors,
-          stack: error.stack
-        });
-      } else {
-        app.log.debug(`Found no dependents ${ name }`);
-      }
+Feedsme.prototype.triggerDep = async function triggerDep({ env, dep, releaseLine, pkg }) {
+  const { name, version } = pkg;
+  const spec = {
+    previousVersion: releaseLine.previousVersion,
+    pkg: name,
+    version
+  };
+  const publish = env === 'dev';
+
+  // If we are a publish, create a release-line
+  // NOTE: this version can be innaccurate if the npm publish was done with
+  // a tag that !== latest. We should handle this case at some point, fetch
+  // package for version or something
+  if (publish) {
+    this.log.info('create release-line', spec);
+    await this.release.create({
+      previousVersion: releaseLine.previousVersion,
+      pkg: name,
+      version
+    });
+  }
 
-      return done(error);
-    }
+  this.log.debug(`Trying to rebuild the dependents of ${ name }`);
+  if (!dep || (dep && dep.dependents.length === 0))
+    return this.log.debug(`Found no dependents for ${ name }`);
```

Comment:
should you do this check before logging the line right before this if-statement?